tools/ci-trigger/pr.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/go-git/go-git/v5"
+	"github.com/golang/glog"
+	"github.com/google/go-github/v50/github"
+	"google.golang.org/api/cloudbuild/v1"
+	"google.golang.org/protobuf/encoding/prototext"
+
+	mpb "github.com/openconfig/featureprofiles/proto/metadata_go_proto"
+	opb "github.com/openconfig/ondatra/proto"
+)
+
+type pullRequest struct {
+	ID       int
+	CloneURL string
+	HeadSHA  string
+	BaseSHA  string
+	Virtual  []device
+	Physical []device
+
+	repo      *git.Repository
+	localFS   fs.FS
+	localPath string
+}
+
+type device struct {
+	Type             deviceType
+	CloudBuildID     string
+	CloudBuildLogURL string
+	Tests            []functionalTest
+}
+
+type deviceType struct {
+	Vendor        opb.Device_Vendor
+	HardwareModel string
+}
+
+type functionalTest struct {
+	Name        string
+	Description string
+	Path        string
+	DocURL      string
+	TestURL     string
+	Status      string
+	BadgePath   string
+	BadgeURL    string
+}
+
+func (d *deviceType) String() string {
+	return d.Vendor.String() + " " + d.HardwareModel
+}
+
+// CreateBuild creates a GCB build for each of the deviceTypes.
+func (p *pullRequest) CreateBuild(ctx context.Context, buildClient *cloudbuild.Service, storClient *storage.Client, devices []deviceType) error {
```

Comment:
Do all the capitalized functions across all the files need to be exported? I don't see them called by another package, but it could be they have to conform to some interface that I don't see.